AlphaWallet/Market/Coordinators/UniversalLinkCoordinator.swift

```diff
@@ -156,19 +156,35 @@ class UniversalLinkCoordinator: Coordinator {
 
     func completeOrderHandling(signedOrder: SignedOrder)
     {
+        let requiresPaymaster = requiresPaymasterForCurrencyLinks(signedOrder: signedOrder)
         if signedOrder.order.price == 0 {
             self.checkPaymentServerSupportsContract(contractAddress: signedOrder.order.contractAddress) { supported in
                 if supported {
                     self.usePaymentServerForFreeTransferLinks(signedOrder: signedOrder)
-                } else {
+                } else if !requiresPaymaster {
                     self.handlePaidImports(signedOrder: signedOrder)
+                } else {
+                    //Currency links on mainnet/classic/xdai without a paymaster should be rejected for security reasons (front running)
+                    self.showImportError(errorMessage: R.string.localizable.aClaimTokenFailedServerDown())
                 }
             }
         } else {
             self.handlePaidImports(signedOrder: signedOrder)
         }
     }
 
+    private func requiresPaymasterForCurrencyLinks(signedOrder: SignedOrder) -> Bool {
```

Comment:
Here's a version using guard to make it clearer that is the precondition you are looking out for:

```
        guard signedOrder.order.nativeCurrencyDrop else { return false }
        guard signedOrder.order.price == 0 else { return false }

        switch config.server {
        case .xDai, .classic, .main:
            return true
        default:
            return false
        }
```

Since there's a switch based on `config.server`, we can actually move that as a property into `RPCServer` too, like how `RPCServer.name` works